File: artifacts/lodepng/translate_without_break_down/src/lode_p_n_g_encoder_settings.rs
use crate::*;

#[derive(Clone)]
pub struct LodePNGEncoderSettings {
    pub zlibsettings: LodePNGCompressSettings,
    pub auto_convert: u32,
    pub filter_palette_zero: u32,
    pub filter_strategy: LodePNGFilterStrategy,
    pub predefined_filters: Option<Vec<u8>>,
    pub force_palette: u32,
    pub add_id: u32,
    pub text_compression: u32,
}

File: artifacts/lodepng/decompose/src/lode_p_n_g_encoder_settings.c
typedef struct LodePNGEncoderSettings
{
  LodePNGCompressSettings zlibsettings;
  unsigned auto_convert;
  unsigned filter_palette_zero;
  LodePNGFilterStrategy filter_strategy;
  const unsigned char *predefined_filters;
  unsigned force_palette;
  unsigned add_id;
  unsigned text_compression;
} LodePNGEncoderSettings
/*** DEPENDENCIES:
struct LodePNGCompressSettings
{
  unsigned btype;
  unsigned use_lz77;
  unsigned windowsize;
  unsigned minmatch;
  unsigned nicematch;
  unsigned lazymatching;
  unsigned (*custom_zlib)(unsigned char **, size_t *, const unsigned char *, size_t, const LodePNGCompressSettings *);
  unsigned (*custom_deflate)(unsigned char **, size_t *, const unsigned char *, size_t, const LodePNGCompressSettings *);
  const void *custom_context;
}
----------------------------
typedef enum LodePNGFilterStrategy
{
  LFS_ZERO = 0,
  LFS_ONE = 1,
  LFS_TWO = 2,
  LFS_THREE = 3,
  LFS_FOUR = 4,
  LFS_MINSUM,
  LFS_ENTROPY,
  LFS_BRUTE_FORCE,
  LFS_PREDEFINED
} LodePNGFilterStrategy
----------------------------
***/


